src/ConstructiveSolidGeometry/IO.jl

```diff
@@ -1,40 +1,52 @@
-    "tube" => Cone,
-    "cone" => Cone,
-    "sphere" => Sphere,
+    # "tube" => Cone,
+    # "cone" => Cone,
+    # "sphere" => Sphere,
     "box" => Box,
-    "torus" => Torus,
-    "TriangularPrism" => TriangularPrism,
-    "SquarePrism"  => SquarePrism,
-    "PentagonalPrism" => PentagonalPrism,
-    "HexagonalPrism"  => HexagonalPrism,
-    "union" => CSGUnion,
-    "difference" => CSGDifference,
-    "intersection" => CSGIntersection,
-    "translate" => TranslatedGeometry,
-    "rotate" => RotatedGeometry,
-    "scale" => ScaledGeometry
+    # "torus" => Torus,
+    # "TriangularPrism" => TriangularPrism,
+    # "SquarePrism"  => SquarePrism,
+    # "PentagonalPrism" => PentagonalPrism,
+    # "HexagonalPrism"  => HexagonalPrism,
+    # "union" => CSGUnion,
+    # "difference" => CSGDifference,
+    # "intersection" => CSGIntersection,
 )
 
-function get_geometry_key(::Type{T}, dict::AbstractDict, input_units::NamedTuple)::Tuple{String, Vector{CSGTransformation}} where {T}
-    dict_keys = filter(k -> k in keys(CSG_dict), keys(dict))
-    transformations = sort!(filter(k -> 
-                                (k == "translate" && !any(broadcast(key -> key in keys(CSG_dict), keys(dict["translate"])))) || 
-                                (k == "rotate" && !any(broadcast(key -> key in keys(CSG_dict), keys(dict["rotate"])))), 
-                            collect(dict_keys)), 
-                      rev = true) #this will ensure that the rotation is parsed first and the translation is parsed second
-    primitives = setdiff(dict_keys, transformations)
-    @assert length(primitives) <= 1 "Too many geometry entries in dictionary: $(length(dict_keys))."
-    @assert length(primitives) >= 1 "None of the entries $(keys(dict)) describes a Geometry."
-    first(primitives), broadcast(t -> parse_CSG_transformation(T, dict, CSG_dict[t], input_units), transformations)
+function get_geometry_key(::Type{T}, dict::AbstractDict, input_units::NamedTuple) where {T}
+    g = collect(filter(k -> k in keys(CSG_dict), keys(dict)))
+    @assert !(length(g) > 1) "Too many geometry entries in dictionary: $(length(g))."
+    @assert !(length(g) == 0) "None of the entries $(keys(dict)) describes a Geometry."
+    g[1]
 end
 
 
+function parse_CSG_transformation(::Type{T}, dict::AbstractDict, input_units::NamedTuple) where {T}
+    r = if haskey(dict, "rotate")
+        SMatrix{3, 3, T, 9}(parse_rotation_matrix(T, dict["rotate"], input_units.angle))
+        # We might want to improve this to avoid numerical precision errors. 
+        # For now this is fine. 
+    else
+        one(SMatrix{3, 3, T, 9})
+    end
+    s = if haskey(dict, "scale")
+        error("Not yet implemented")
+    else
+        ones(SVector{3,T})
+    end
+    t = if haskey(dict, "translate")
+        parse_translate_vector(T, dict["translate"], input_units.length)
+    else
+        zero(CartesianVector{T})
+    end
+    (scale = s, rotation = r, translation = t)
```

Comment:
By doing this, we will not allow for the previously supported format of:
```yaml
translate:
  z: 1.0
  tube:
    r: 1.0
    z: 1.0
```
i.e. formats where primitives are part of `translate` etc.  The primitives would just be ignored here.

Reply:
Ah, right. Yes, this has to be changed.

Reply:
In what way? Do we still want to support the old syntax or adapt the config files?

Reply:
Nope. We want nested read-in like you posted it. 
This is now (https://github.com/JuliaPhysics/SolidStateDetectors.jl/pull/163/commits/562369ce5c4aab580b718dd7186dc86d8a404741) possible again inside the "geometry"-fields of contact and so on. 
Outside, on the detector level in the configuration file, it is still possible to define translation and rotation on the same level.
It will always rotate first and than translate. 
I hope I did everything correct.